File: 152-MaximumProductSubarray/152-MaximumProductSubarray.cpp
// Last updated: 04/04/2026, 23:36:28
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n = nums.size();
        int ans = INT_MIN;
        int prefix = 1;
        int suffix = 1;
        for(int i=0;i<n;i++) {
            if(prefix==0) prefix = 1;
            if(suffix==0) suffix = 1;
            prefix *= nums[i];
            suffix *= nums[n-i-1];
            ans = max(ans,max(prefix,suffix));
        }
        return ans;
    }
};

File: 152-MaximumProductSubarray/152-MaximumProductSubarray.java
// Last updated: 04/04/2026, 23:36:32
class Solution {
    public int maxProduct(int[] nums) {
        int n = nums.length;
        int ans = Integer.MIN_VALUE;
        for(int i=0;i<n;i++) {
            int product = 1;
            for(int j=i;j<n;j++) {
                product *= nums[j];
                if(ans < product) ans = product;
            }
        }
        return ans;
    }
}